h/traversal/group_membership.py

```diff
@@ -10,50 +10,74 @@
 class GroupMembershipContext:
     group: Group
     user: User
-    membership: GroupMembership | None
+    membership: GroupMembership
+
+
+@dataclass
+class AddGroupMembershipContext:
+    group: Group
+    user: User
+    new_roles: list[GroupMembershipRoles] | None
```

Comment:
It's unfortunate to have to type-annotate this as `| None` here, but `AddGroupMembershipContext`'s are initially created with `new_roles=None` and then `new_roles` is set later so :man_shrugging: